docs: update the perception design document and the interface document (#499)
autowarefoundation/autoware-documentation: docs/design/autoware-architecture/perception/index.md

The Perception Component consists of the following sub-components:

- **Object Recognition**: Recognizes dynamic objects surrounding the ego vehicle in the current frame, objects that were not present during map creation, and predicts their future trajectories. This includes:
  - Pedestrians
  - Cars
  - Trucks/Buses
  - Bicycles
  - Motorcycles
  - Animals

>   - Traffic Cones

Reviewer: NITS: It is advisable to standardize the format by either capitalizing the initial letter (Cars) or using lowercase (trees)."

>   - Traffic Cones

Reviewer: NITS: It is advisable to standardize the format by either capitalizing the initial letter (Cars) or using lowercase (trees).

Author: https://github.com/autowarefoundation/autoware-documentation/pull/499/commits/8454221efbf20ee523718608c51886e80beb7471

>   - Fallen Objects, Flying Objects, etc: Items such as cardboard, drums, wood, etc., either falling or airborne.

Reviewer: ```suggestion
  - Road debris: Items such as cardboard, oil drums, wood, etc., either dropped on the road or floating in the air, posing challenges for autonomous vehicles.
```

Author: https://github.com/autowarefoundation/autoware-documentation/pull/499/commits/8454221efbf20ee523718608c51886e80beb7471

- **Obstacle Segmentation**: Identifies point clouds originating from obstacles, including both dynamic objects and static obstacles that requires the ego vehicle either steer clear of them or come to a stop in front of the obstacles.
  - This includes:
    - all dynamic objects (as listed above)
    - curbs/bollards
    - barriers
    - trees

>     - walls/buildings

Reviewer: NITS: It is advisable to standardize the format by either capitalizing the initial letter (Cars) or using lowercase (trees).

>     - etc

Reviewer: ```suggestion
```

Author: https://github.com/autowarefoundation/autoware-documentation/pull/499/commits/8454221efbf20ee523718608c51886e80beb7471

  - This does not include:
    - grass
    - water splashes
    - smoke/vapor
    - newspapers

>     - plastic bags

Reviewer: NITS: It is advisable to standardize the format by either capitalizing the initial letter (Cars) or using lowercase (trees).

>     - etc

Reviewer: ```suggestion
```

Author: https://github.com/autowarefoundation/autoware-documentation/pull/499/commits/8454221efbf20ee523718608c51886e80beb7471
